### PR TITLE
Plans Page: Fix floating point issue for Euro currency

### PR DESCRIPTION
**Pull request by enejb**
Currently we don't take account different currencies in the Plans Price Display.

Before:
Screen Shot 2019-11-21 at 10 47 49 AM

After:
Screen Shot 2019-11-21 at 10 42 38 AM

#### Changes proposed in this Pull Request:
* Add a new function that will allow us to format the prices in a more internationalized way.

#### Testing instructions:
* With prices in euros go to the plans page. 
* Does the price look as expected.

### CODE REVIEW

File: _inc/client/plans/single-product-backup.jsx [hunk 1]
```diff
@@ -61,24 +63,54 @@ export function SingleProductBackup( { products, upgradeLinks } ) {
...
+				currencySymbol={ currencySymbol }
+				currencyCode={ currencyCode }
+				backupPlanPrices={ backupPlanPrices }
+			/>
 		</div>
 	);
 }
 
-export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+function formatCurrencyNumber( number, code ) {
```
Comment (delawski) [inline _inc/client/plans/single-product-backup.jsx]: Do we really need to copy this over from `@automattic/format-currency`? If we imported it at the first place like:

```js
import formatCurrency from '@automattic/format-currency';
```

then we could use it as noted in the other comment.

File: _inc/client/plans/single-product-backup.jsx [hunk 2]
```diff
@@ -61,24 +63,54 @@ export function SingleProductBackup( { products, upgradeLinks } ) {
...
+export function PlanPriceDisplay( { backupPlanPrices, currencySymbol, currencyCode } ) {
 	const dailyBackupYearlyPrice = backupPlanPrices.jetpack_backup_daily.yearly;
 	const dailyBackupMonthlyPrice = backupPlanPrices.jetpack_backup_daily.monthly;
 
 	const realtimeBackupYearlyPrice = backupPlanPrices.jetpack_backup_realtime.yearly;
 	const realtimeBackupMonthlyPrice = backupPlanPrices.jetpack_backup_realtime.monthly;
 
-	const fullDailyBackupYearlyCost = dailyBackupMonthlyPrice * 12;
-	const fullRealtimeBackupYearlyCost = realtimeBackupMonthlyPrice * 12;
+	const fullDailyBackupYearlyCost = formatCurrencyNumber(
```
Comment (delawski) [inline _inc/client/plans/single-product-backup.jsx]: I guess we could use the `formatCurrency` function directly from '@automattic/format-currency' here. We could tell it to drop the decimals and show the currency symbol only for the first price. Eventually we could land with something like this:

```js
<div className="discounted-price__price">
	{ __( '%(lowPrice)s-%(highPrice)s', {
		args: {
			lowPrice: formatCurrency(
				dailyBackupMonthlyPrice * 12,
				currencyCode,
				{
					precision: 0,
				}
			),
			highPrice: formatCurrency(
				realtimeBackupMonthlyPrice * 12,
				currencyCode,
				{
					precision: 0,
					symbol: '',
				}
			),
		},
		comment:
			"Describes how much a plan will cost per year. %(currencySymbol) is the currency symbol of the user's locale (e.g. $). %(planPrice) is the cost of a plan (e.g. 20).",
	} ) }
</div>
```

The same principle would apply to the non-discounted price.

File: _inc/client/plans/single-product-backup.jsx [hunk 3]
```diff
@@ -61,24 +63,54 @@ export function SingleProductBackup( { products, upgradeLinks } ) {
...
 }
 
-function SingleProductBackupHeader( { currencySymbol, backupPlanPrices } ) {
+function SingleProductBackupHeader( { currencySymbol, currencyCode, backupPlanPrices } ) {
 	return (
 		<div className="single-product-backup__header-container">
 			<h3 className="single-product-backup__header-title">{ __( 'Jetpack Backup' ) }</h3>
-			<PlanPriceDisplay currencySymbol={ currencySymbol } backupPlanPrices={ backupPlanPrices } />
+			<PlanPriceDisplay
+				currencySymbol={ currencySymbol }
```
Comment (delawski) [inline _inc/client/plans/single-product-backup.jsx]: If we use `formatCurrency` function from `@automattic/format-currency` we won't need the `currencySymbol` - the `currencyCode` would suffice.

**Review by delawski — CHANGES_REQUESTED** [review]
Thank you for your work on this.

I think we could use `formatCurrency` function from `@automattic/format-currency` directly instead of copying the function code to our component here. I left some comments in the code. Please have a look.

**Review by jeherve — COMMENTED** [review]
I am not sure how we are getting in this situation, but then again I am getting prices in HUF on my test site, so I'm probably not the best person to review this :)

Noting that price displays were discussed in #14060 as well.

**Comment by delawski** [comment]
Closing in favour of https://github.com/Automattic/jetpack/pull/14089

**Pull request closed without merging**